### PR TITLE
feat(api): onNewTrack, onNewView

### LINKED ISSUE #926: onNewView/onNewTrack functionality
Create an event that is dispatched when a new view/track has been created to get the data and other information/e.g. dimensions, specification etc. about the track/view.

See also view API of vega: https://vega.github.io/vega/docs/api/view/

### PR DESCRIPTION
**Pull request by etowahadams**
Fix #926
Toward #

## Change List
 - Add `onNewTrack` API
 - Add `onNewView` API

```
// New Track
gosRef.current.api.subscribe('onNewTrack', (type, eventData) => {
    console.warn(eventData.id, eventData.shape);
});
// New View
gosRef.current.api.subscribe('onNewView', (type, eventData) => {
     console.warn(eventData.id, eventData.shape);
});
```

## Checklist
 - [ ] Ensure the PR works with all demos on the online editor
 - [ ] Unit tests added or updated
 - [ ] Examples added or updated
 - [ ] [Documentation](https://github.com/gosling-lang/gosling-website) updated (e.g., added API functions)
 - [ ] Screenshots for visual changes (e.g., new encoding support or UI change on Editor)

### CODE REVIEW

**Comment by etowahadams** [comment]
I'll add documentation once we decide that this is the final API we want to settle on

File: src/core/gosling.schema.ts [hunk 1]
```diff
@@ -242,6 +242,18 @@ export interface GenomicPosition {
...
+    id: string;
+    /** The shape of the source track */
+    shape: BoundingBox | CircularTrackShape;
+}
+interface OnNewViewEventData {
+    /** Source visualization ID, i.e., `track.id` */
+    id: string;
+    /** The shape of the source track */
+    shape: BoundingBox;
+}
```
Comment (etowahadams) [inline src/core/gosling.schema.ts]: Currently onNewTrack and onNewView simply return the ID and the shape

Comment (sehilyi) [reply src/core/gosling.schema.ts]: I think we can also add `spec` as what @ThHarbig provided as an example (#926). We can reuse `TrackApiData` and `ViewApiData` for this.

@ThHarbig If you could share with us use cases, that would be helpful for determining what information we want to return.

Comment (sehilyi) [reply src/core/gosling.schema.ts]: Thinking more about this, I think we can just return an ID of a track or a view when it is mounted and then let users use another API function to get relevant information, such as the bounding box and spec:

```ts
const trackIdOfInterest = 'track-1';
gosRef.current.api.subscribe('onNewTrack', (type, eventData) ⇒ {
   if(eventData.id === trackIdOfInterest) {
      // 'track-1' has been mounted, so let's get the related data
      console.log(gosRef.current.api.getTrack(trackIdOfInterest); // get the bounding box and the spec
   }
});
```

I think this is cleaner as the information we return is not duplicated across API functions.

Comment (etowahadams) [reply src/core/gosling.schema.ts]: That makes a lot of sense to me!

**Comment by etowahadams** [comment]
Another nuance to address: If a view does not have a user-defined ID in the schema, then an ID will be auto-generated. The auto-generation happens every time the spec is compiled. So for every view without a user-defined ID will trigger onNewView when the spec is re-compiled, since there is a new ID compared to the set of previous view IDs. Is this the expected behavior?

**Comment by sehilyi** [comment]
It is not intended, and I suspect that this might be an issue of the editor and not the gosling.js. When I print something inside the `compiled` callback function in the `gosling-react-template` repo, which is a simpler example of using Gosling.js, I see the message printed only once.

```diff
				experimental={{ reactive: true }}
+				compiled={() => console.log('Compiled')}
			/>
```

https://github.com/gosling-lang/gosling-react-example/blob/a695b2ba532e35bc2f9e813a8aa1e385da8bb636/src/example/Simple.jsx#L30

I think as the editor becomes complex, we do not handle the React rendering cycle well, resulting in calling GoslingComponent multiple times with the same user spec.

**Review by sehilyi — COMMENTED** [review]
Other than the event data type I commented on, this looks great to me. Since Theresa might not be available, we could merge this without her review for now and make improvements later when we can get input from her.

**Comment by etowahadams** [comment]
Made the API only return the track/view ID! Going to merge now

**Pull request merged**